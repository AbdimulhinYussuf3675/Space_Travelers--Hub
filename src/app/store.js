import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from 'features/Rockets/rockets';
import missionReducer from '../features/Mission/mission';

export default configureStore({
  reducer: {
    books: missionReducer,
    rocket: rocketReducer,
  },
});
