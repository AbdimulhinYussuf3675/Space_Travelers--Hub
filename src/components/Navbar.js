import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from 'styles/Nav.modules.css';

const Navbar = () => (
  <div className={styles.navBar}>
    <h1>
      <NavLink to="/" className={styles.logo}>
        Space Travelers
      </NavLink>
    </h1>
    <nav>
      <ul className={styles.navLinks}>
        <li className={styles.links}>
          <NavLink to="/" className={styles.link}>
            Rockets
          </NavLink>
        </li>
        <li className={styles.links}>
          <NavLink to="/mission" className={styles.link}>
            Missions
          </NavLink>
        </li>

      </ul>
    </nav>
  </div>
);

export default Navbar;
