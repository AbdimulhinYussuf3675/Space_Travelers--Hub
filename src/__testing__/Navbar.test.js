import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from 'app/store';
import Navbar from 'components/Navbar';

describe('Navbar test', () => {
  test('check if navigation rendered', () => {
    const navbar = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
