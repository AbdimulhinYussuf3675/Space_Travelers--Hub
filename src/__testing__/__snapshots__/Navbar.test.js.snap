// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test check if navigation rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navBar"
      >
        <div
          class="logo"
        >
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            <img
              alt="travelers logo "
              src="logo.png"
            />
            <h1>
              Space Rockets
            </h1>
          </a>
        </div>
        <ul
          class="navLinks"
        >
          <li
            class="links"
          >
            <a
              aria-current="page"
              class="link active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/mission"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="link"
              href="/Profile"
            >
              Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navBar"
    >
      <div
        class="logo"
      >
        <a
          aria-current="page"
          class="link active"
          href="/"
        >
          <img
            alt="travelers logo "
            src="logo.png"
          />
          <h1>
            Space Rockets
          </h1>
        </a>
      </div>
      <ul
        class="navLinks"
      >
        <li
          class="links"
        >
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/mission"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="link"
            href="/Profile"
          >
            Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
